### PR TITLE
Improve task/launches dashboard

### PR DESCRIPTION
**Pull request by Wintxer**

### CODE REVIEW

File: webapp/src/components/JobsListing/JobTableView.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+interface PropType {
+    content: Array<JobsType>;
+}
+
+const JobTableView = (props: PropType) => {
+  const { content } = props;
+  const [t] = useTranslation();
+  const classes = useStyles();
+  const [sorted, setSorted] = useState<string>();
+  const [type, setType] = React.useState('');
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx]: replace by useState

Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx:50]: replace by useState

File: webapp/src/components/JobsListing/JobTableView.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          labelId="jobsView-select-outlined-label"
+          id="jobsView-select-outlined"
+          value={type}
+          onChange={handleChange}
+          label={t('jobs:typeFilter')}
+        >
+          {filterList.map((item) =>
+            (
+              <MenuItem value={item === 'all' ? '' : item} key={item}>
+                {t(`jobs:${item}`)}
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx:90]: Nothing is displayed when you click on "all". Why not something like: 
{ item === 'all' ? t('main:all') : t(`jobs:${item}`)}

Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx:90]: Nothing is displayed when you click on "all". Why not something like: 
{ item === 'all' ? t('main:all') : t(`jobs:${item}`)}

File: webapp/src/components/JobsListing/JobTableView.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        <Select
+          labelId="jobsView-select-outlined-label"
+          id="jobsView-select-outlined"
+          value={type}
+          onChange={handleChange}
+          label={t('jobs:typeFilter')}
+        >
+          {filterList.map((item) =>
+            (
+              <MenuItem value={item === 'all' ? '' : item} key={item}>
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx:89]: Currently nothing is displayed when you click on "all". 
Why not value={item} and initialize type whit 'all'

Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx]: Currently nothing is displayed when you click on "all". 
Why not value={item} and initialize type whit 'all' ?

File: webapp/src/components/JobsListing/JobTableView.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  const [type, setType] = React.useState('');
+  const [currentContent, setCurrentContent] = useState<JobsType[]>();
+
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setType(event.target.value as string);
+  };
+
+  useEffect(() => {
+    if (content) {
+      if (type) {
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx]: tyype !== 'all'

Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx:59]: tyype !== 'all'

File: webapp/src/components/JobsListing/JobTableView.tsx (hunk 1 above)
Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx]: Initialize with 'all'

Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx:50]: Initialize with 'all'

File: webapp/src/components/JobsListing/JobTableView.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                    <ArrowDropDownIcon className={classes.filterIcon} onClick={() => setSorted(undefined)} color="primary" />)}
+                </Box>
+              </TableCell>
+              <TableCell align="right">
+                {t('jobs:action')}
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {!sorted && (sorted !== 'date') && currentContent ? currentContent.sort((a, b) => (moment(a.date).isAfter(moment(b.date)) ? -1 : 1)).map((row) => (
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx]: Why not initialize currentContent with [ ] (void array)

Then with less code : 
{ currentContent.sort((a,b) =>  {
 HERE USE YOUR CONDITION  ----> !sorted && (sorted !== 'date')
} ).map(...)
}

Comment (3lbanna) [inline webapp/src/components/JobsListing/JobTableView.tsx:118]: With less code : 
{ currentContent && currentContent.sort((a,b) =>  {
 HERE USE YOUR CONDITION  ----> !sorted && (sorted !== 'date')
} ).map(...)
}

File: webapp/src/components/JobsListing/index.tsx [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+  const renderStatus = (job: LaunchJob) => {
+    let color = theme.palette.grey[400];
+    if (job.status === 'success') {
+      color = theme.palette.success.main;
+    } else if (job.status === 'failed') {
+      color = theme.palette.error.main;
+    } else if (job.status === 'running') {
+      color = theme.palette.warning.main;
+    }
+    return (<Box className={classes.dot} style={{ backgroundColor: color }} />);
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx:160]: You can use a material ui dot icon.
Ex: FiberManualRecordIcon in '@material-ui/icons/FiberManualRecord';

Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx]: You can use a material ui dot icon.
Ex: FiberManualRecordIcon in '@material-ui/icons/FiberManualRecord';

File: webapp/src/components/JobsListing/index.tsx [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+      tasks.forEach((task) => {
+        subscribeChannel(WsChannel.TASK + task.id);
+      });
+      return () => {
+        tasks.forEach((task) => {
+          unsubscribeChannel(WsChannel.TASK + task.id);
+        });
+      };
+    }
+    return () => { /* noop */ };
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx:230]: remove

File: webapp/src/components/JobsListing/index.tsx [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+              {convertUTCToLocalTime(job.completionDate)}
+            </>
+          )}
+        </Box>
+      </Box>
+    ),
+    action: (
+      <Box className={classes.actions}>
+        <Box className={classes.actionButton}>
+          {job.status === 'running' ? <BlockIcon className={classes.blockIcon} onClick={() => setOpenConfirmationModal(job.id)} /> : <Box />}
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx]: Add a tooltip with an explicit title for actions, and not just an Icon

File: webapp/src/components/JobsListing/index.tsx [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+      <Box className={classes.title}>
+        {download.name}
+      </Box>
+    ),
+    dateView: (
+      <Box className={classes.dateblock}>
+        {`(${t('downloads:expirationDate')} : ${convertUTCToLocalTime(download.expirationDate)})`}
+      </Box>
+    ),
+    action: (
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx:300]: Add tooltip for actions

Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx:294]: Add tooltip for actions

File: webapp/src/components/JobsListing/index.tsx [hunk 10]
```diff
@@ -0,0 +1,374 @@
...
+          {task.completion_date_utc && (
+            <>
+              <FontAwesomeIcon className={classes.dateicon} icon="calendar-check" />
+              {convertUTCToLocalTime(task.completion_date_utc)}
+            </>
+          )}
+        </Box>
+      </Box>
+    ),
+    action: (
```
Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx:349]: Same

Comment (3lbanna) [inline webapp/src/components/JobsListing/index.tsx:333]: Same

File: webapp/src/components/JobsListing/index.tsx (hunk 7 above)
Comment (Wintxer) [reply webapp/src/components/JobsListing/index.tsx:230]: ESlint is expecting a return in this useEffect, how should I handle that if I remove the return?

**Review by 3lbanna — APPROVED** [review]

**Pull request merged**